### PR TITLE
OCPBUGS-18586: Fix OVN SNATing on GR by enabling gateway_mtu on rtoe port of GR

### PR DESCRIPTION
**Pull request by tssurya**
minor conflict outlined in 1st commit.
/assign @trozet

### CODE REVIEW

File: go-controller/pkg/ovn/gateway_init.go [hunk 1]
```diff
@@ -470,19 +475,37 @@ func (oc *Controller) addExternalSwitch(prefix, interfaceID, nodeName, gatewayRo
...
 		ExternalIDs: map[string]string{
 			"gateway-physical-ip": "yes",
 		},
 		Networks: externalRouterPortNetworks,
 		Name:     externalRouterPort,
+		Options:  options,
 	}
 	logicalRouter := nbdb.LogicalRouter{Name: gatewayRouter}
 
+<<<<<<< HEAD
```
Comment (zhaozhanqi) [inline go-controller/pkg/ovn/gateway_init.go:495]: @tssurya conflict here.

**Comment by anuragthehatter** [comment]
verified repro steps and upgrade. Looks good on cluster bot build

/label qe-approved

**Comment by trozet** [comment]
migration jobs are pretty much perma fail since mid august due to https://issues.redhat.com/browse/OCPBUGS-17391

overriding

/override ci/prow/e2e-aws-ovn-local-to-shared-gateway-mode-migration
/override ci/prow/e2e-aws-ovn-shared-to-local-gateway-mode-migration

**Comment by abhat** [comment]
@anuragthehatter can you move the dependent bug to verified please?

**Pull request merged**